topdown/time.go

```diff
@@ -124,6 +124,73 @@ func builtinAddDate(bctx BuiltinContext, operands []*ast.Term, iter func(*ast.Te
 	return iter(ast.NewTerm(ast.Number(int64ToJSONNumber(result.UnixNano()))))
 }
 
+func builtinDiff(a, b ast.Value) (ast.Value, error) {
+	t1, err := tzTime(a)
+	if err != nil {
+		return nil, err
+	}
+	t2, err := tzTime(b)
+	if err != nil {
+		return nil, err
+	}
+
+	// The following implementation of this function is taken
+	// from https://github.com/icza/gox licensed under Apache 2.0.
+	// The only modification made is to variable names.
+	//
+	// For details, see https://stackoverflow.com/a/36531443/1705598
+	//
+	// Copyright 2021 icza
+	// BEGIN REDISTRIBUTION FROM APACHE 2.0 LICENSED PROJECT
+	if t1.Location() != t2.Location() {
+		t2 = t2.In(t1.Location())
+	}
+	if t1.After(t2) {
```

Comment:
Just a reflection, do we want this? Comes back to the question if a diff function always should return an absolute value? I guess if you as you want to know if its the future or past its easier to compare a and b directly before calling this.

Reply:
Probably a good idea. Any suggestion on what we could return if a < b?